app/adapters/signal_adapter/outbound/text.rb

```diff
@@ -19,14 +19,7 @@ def perform(recipient:, text:)
         response = Net::HTTP.start(url.host, url.port) do |http|
           http.request(request)
         end
-        response.value # may raise exception
-      rescue Net::HTTPClientException => e
```

Comment:
If you remove the `rescue` then you will not even get to `handle_response` if an exception is raised.

Reply:
Ok, I see from the comment now that `response.value` may raise the error and it was also removed, nevermind. But I find the behaviour surprising.

Reply:
Yeah, I find the behavior as it exists surprising, to be honest. We call `response.value` because if it is an unsuccessful response calling `.value` on it raises an exception, we even explain that the only purpose for line `22` is that it may raise an exception.

I think a better way to handle it is to either check the class of the response, or check its code to determine whether the response was successful, or not.